clients/ts/FunctionalTests/ts/ConnectionTests.ts

```diff
@@ -1,6 +1,9 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+// This code uses a lot of `.then` instead of `await` and TSLint doesn't like it.
+// tslint:disable:no-floating-promises
```

Comment:
Whats wrong with using await?

And wasn't this option enabled partially because we wanted to find possibly flaky tests because of floating promises?

Reply:
These tests need to be fully converted to use promises properly. They currently use `done` instead. I didn't want to go through and completely rework them (including adding the necessary helpers so jasmine works) in this PR.